docs/common-sdk-requirements.md

```diff
@@ -0,0 +1,43 @@
+# Requirements for the Common SDK infrastructure
+The purpose of this doc is to have a place to list and discuss all requirements to the new modular SDK infrastructure (also informally called "the Common SDK infrastructure") before creating an official design.
+
+The information listed here was primarily sourced in a few meetings/discussions/whiteboardings we had before.
+
+## Terminology
+* _Common SDK_ is the new modular SDK. 
+* _Service_ is a single building unit (module) of the new modular SDK. TODO: please let me know if another term is desired for this.
+
+## Requirements
+### Requirements to the SDK
+* Greater degree of flexibility and scalability in development, maintenance, customization and extensibility due to:
+    * Non-monolithic, modular structure
+    * Smaller building blocks (services) that can be worked on independently
+    * High configurability (configuration-driven design)
+* Configurable composition:
+    * Services can be selected
+    * Services can be optional
+    * Services can be required
+    * Composition rules/restrictions apply based on each service dependencies and configuration
+    * Configuration errors must be prevented/communicated at the time of configuring
+* Ability to tailor SDK to specific customer needs
+* Can be composed and configured by a customer
+* Trackable SDK generation and download process to provide acceptable customer experience
```

Comment:
We'd like to remember player profile for SDK re-generation. Currently we do not require folks to be logged in to download SDKs - are we going to need that in future ? Might create a barrier to entry - if not how do we remember profiles ? Can they save their initial configuartion and re-use that ?

Reply:
Option 1: We require registration at PlayFab to be able to use the new modular SDK platform (then everything will be saved in user's PlayFab account, including SDK configurations, SDK generation/download process tracking, etc)

Option 2: We could potentially provide the functionality of service selection and SDK configuration for anonymous users (public web access) and allow them to either upload/download a configuration profile as a JSON file or maybe store it in web cookies (though it may have some length limitation). Without the actual registration at PlayFab though they won't be able to do much with downloaded SDK (but it may be ok).

What do you guys think?

Reply:
Let's capture this as a requirement and say that we'll address this later when we hot builds the sdks. But let's capture your and Paul's thoughts as an appendix(?) or any markdowny way you deem fit.

There are a few things we should avoid:
1. Making login mandatory to get an SDK - this just puts more barrier to entry for onboarders.
2. Cookie idea is difficult to implement and it's probabably going to be weeks between subsequent regenerations. Plus now we might need to make it GDPR compliant.
3. A safer bet is informing them via a dialog that either prompts them to login while explaining the benefit or has a "download anyways"
4. I love the idea of giving them a json file they can save and later upload to bypass the web UI selection process if not logged in. We'll need to explore and think about it more though.